src/message_queue.rs

```diff
@@ -6,85 +6,54 @@
 // License, Version 2.0 found in the LICENSE-APACHE file in the root directory
 // of this source tree.
 
+//! The `MessageQueue` type for storing and retrieving a queue of messages.
+//!
+//! [`MessageQueue`] provides a simple means of storing and retrieving messages
+//! associated with a given [`MessageType`]. Messages can be retrieved either
+//! all at once using [`MessageQueue::retrieve_all`] or associated with a given
+//! [`ParticipantIdentifier`] using [`MessageQueue::retrieve`].
+
 use crate::{
     errors::Result,
     messages::{Message, MessageType},
-    protocol::Identifier,
     ParticipantIdentifier,
 };
-use serde::{Deserialize, Serialize};
-use std::{collections::HashMap, fmt::Debug};
-
-#[derive(Debug, Serialize, Deserialize)]
-struct MessageIndex {
-    message_type: MessageType,
-    /// `identifier` corresponds to a unique session identifier.
-    identifier: Identifier,
-}
-
-#[derive(Serialize, Deserialize, Clone)]
-pub(crate) struct MessageQueue(HashMap<Vec<u8>, Vec<Message>>);
+use std::collections::HashMap;
 
-impl Default for MessageQueue {
-    fn default() -> Self {
-        Self::new()
-    }
-}
+/// A type for storing a queue of [`Message`]s by [`MessageType`].
+#[derive(Clone, Default)]
+pub(crate) struct MessageQueue(HashMap<MessageType, Vec<Message>>);
 
 impl MessageQueue {
-    pub(crate) fn new() -> Self {
-        Self(HashMap::new())
-    }
-
-    /// Store a message in the MessageQueue.
-    ///
-    /// `identifier` corresponds to a unique session identifier.
-    pub(crate) fn store(
-        &mut self,
-        message_type: MessageType,
-        identifier: Identifier,
-        message: Message,
-    ) -> Result<()> {
-        let key = Self::get_key(message_type, identifier)?;
-        self.0.entry(key).or_default().push(message);
+    /// Store a message by the given [`MessageType`].
+    pub(crate) fn store(&mut self, message_type: MessageType, message: Message) -> Result<()> {
+        self.0.entry(message_type).or_default().push(message);
```

Comment:
thought: The `Message` has a `MessageType` field. In the interest of misuse-resistant internal APIs, I'd rather access that directly, rather than passing it as a separate parameter and risk having a type mismatch. However, I see that the only place this is called does exactly that.

Reply:
Good catch!